src/afrolabs/components/kafka.clj

```diff
@@ -742,18 +742,21 @@
           [_ _]
         (csp/close! caught-up-ch))
 
-      IConsumerPostInitHook
-      (post-init-hook
-          [_ consumer]
-        ;; We save the end offsets at the start, so we can compare later if we've caught up
-        ;; If this times out, the exception will be bubbled up.
-        (reset! end-offsets-at-start
-                (consumer-end-offsets consumer :timeout-value :throw)))
-
       IPostConsumeHook
       (post-consume-hook
           [_ consumer _consumed-records]
+        ;; we have to store the endOffsets, after the first consume (ie after partition assignment)
+        ;; if we have a value in this atom, just keep it
+        ;; otherwise query the end offsets and store it
+        (swap! end-offsets-at-start
+               (fn [old-value]
+                 (if old-value
+                   old-value
+                   (consumer-end-offsets consumer :timeout-value :throw))))
+
```

Comment:
It's very late in the game to catch this bug.

Before, we would attempt to read the endOffsets for every assigned partition, but do this call after the consumer was initialized, but before the first .poll was called. This results in not storing any endOffsets, which meant this ktable started returning data way before it has consumed enough records.

With this change, the ktable will store the endOffsets only after the first .poll / consume, and wait much longer before returning the first data with `deref`.